src/pudl_archiver/depositors/fsspec.py

```diff
@@ -144,34 +217,83 @@ async def open_draft(self) -> "FsspecDraftDeposition":
             settings=self.settings,
             api_client=self.api_client,
             dataset_id=self.dataset_id,
+            # When we open a new draft we assume it starts with all files from previous version
+            resources_in_draft={
+                fname: self.deposition.get_deposition_path(DepositionStatus.PUBLISHED)
+                / fname
+                for fname in await self.list_files()
+            },
         )
 
 
 class FsspecDraftDeposition(DraftDeposition):
-    """Represents draft version of fsspec deposition."""
+    """Represents draft version of fsspec deposition.
+
+    This class tracks the state of the deposition in the ``resources_in_draft`` field.
+    This field is a dictionary that maps filenames to a ``UPath`` that points to the actual
+    file in the deposition. If we are creating a new version of an existing archive,
+    we will populate this with the existing resources from the previous version. In
+    this case, the paths will all point to the files in the 'published' directory.
+    As we add new files, the paths will point to the 'draft' directory. If we update
+    an existing file, we will remove the 'published' path from ``resources_in_draft``
+    and replace it with a path pointing to the updated file in the 'draft' directory.
+    When this happens, we will also add the 'published' version of the file to the
+    ``files_to_delete`` field. This allows us to avoid actually deleting the files
+    until a run is complete and we call the ``publish`` method. If a run fails, all
+    of the original files will remain in the 'published' directory.
```

Comment:
So we will update all of the files, or none of the files, rather than incrementally updating individual files as they succeed?

But then there's a "retry failed" option, where you can try again on just the files that had issues (using a structured failure report from the run that had problems), so we aren't always starting over from scratch just because of a little network glitch?